README.md

```diff
@@ -99,3 +99,52 @@ optional additional logging levels via the `--v=<level>` flag. In general:
 * `--v=3`: log additional data when logging returned values, usually entire go structs
 * `--v=5`: log every function call, including those called very frequently
+## How It Works
+
+The Kubernetes CCM for Packet deploys as a `Deployment` into your cluster with a replica of `1`. It provides the following services:
+
+* lists available zones, returning Packet regions
+* lists and retrieves instances by ID, returning Packet servers
+* manages load balancers
+
+### Load Balancers
+
+Packet does not offer managed load balancers like [AWS ELB](https://aws.amazon.com/elasticloadbalancing/) or [GCP Load Balancers](https://cloud.google.com/load-balancing/).
+Instead, Packet uses BGP and [metallb](https://metallb.universe.tf) to provide the _equivalence_ of load balancing, without requiring an additional
+managed service (or hop).
+
+You can disable the load balancer in the configuration, which will prevent it from being deployed. To do so,
+add the flag `--disable-load-balancer=true` to the CCM command-line options. `--disable-load-balancer`
+defaults to `false`, thus enabling the load balancer.
+
+The Packet CCM, when the load balancer feature is not disabled, does the following.
+
+Upon start, CCM will begin a separate routine that will do the following on start and ongoing:
+
+1. Ensure that the project has BGP enabled
+1. Ensure that each node in the cluster has BGP enabled
+1. Ensure MetalLB is deployed to the cluster, using an ASN of `65000`, while the top-of-rack (ToR) routers are `65530`
+1. Check for node changes, and based on them, configure the single metallb `ConfigMap`:
+   * New node has been added: add the top-of-the-rack (ToR) router peer IP address for that node
+   * Node has been removed: remove the top-of-the-rack (ToR) router peer IP address for that node
+1. Synchronize the list of `Service` of `type=LoadBalancer` to current IPs.
+
+The synchronization logic is as follows. On each loop, compare the list of:
+
+* `Service` of `type=LoadBalancer`
+* Facility-specific `/32` IP address reservations tagged with `"packet-ccm-auto"`
+* Facility-specific `/32` IP address reservation requests tagged with `"packet-ccm-auto"`
+
+The process is as follows goes through each `Service` and the tagged IP address reservations.
+
+1. If there are `Service` with IP pending, and there are available tagged Reservations, add them to the `ConfigMap`
+1. If there remain `Service` with IP pending, and there are no available tagged Reservation or Request, create the request; if it responds immediately, add them
+1. At this point, `count(Service) <= count(Reservation+Request)`
+1. If `count(Requests+Reservations) > count(Service)`, delete Requests until the count is the same, or there are no more Requests
+1. If `count(Reservations) > count(Service)`:
+   1. find unused IPs in the `ConfigMap` by comparing with `Service`s and remove them from the `ConfigMap`
+   1. delete Reservations for those IPs
```

Comment:
Do we think that auto cleaning up the unused IPs is a good thing? I can see situations where you want to delete the service but keep the IP since that is something that you cannot usually change quickly like in DNS.

Reply:
I think so, yes, if we can tag them appropriately. DNS may be quick to change, but it actually is a lot slower to propagate (especially if you have a longer TTL, or clients that don't respect it). 

The other consideration is cost. Customers pay for those IPs, and so if they are created automatically when a `Service` is created, they should be deleted when the `Service` is deleted.

I definitely am open to adding an option later to disable it, though. :-)